lib/pool.py

```diff
@@ -136,3 +136,11 @@ def install_custom_uefi_certs(self, auths):
             host.ssh(['secureboot-certs', 'install'] + params)
         finally:
             host.ssh(['rm', '-f'] + list(auths_dict.values()))
+
+    def eject_host(self, host):
+        master = self.master
+        master.xe('pool-eject', {'host-uuid': host.uuid, 'force': True})
+        wait_for_not(lambda: host.uuid in self.hosts_uuids(), "Wait for host {host} to be ejected of pool {master}.")
+        hosts = self.hosts
```

Comment:
I don't think this is useful, as you only copy the reference.
```
>>> a = [1,2]
>>> b = a
>>> b.append(3)
>>> a
[1, 2, 3]
```

However, AFAICT, in the next line self.hosts will only be written after `[h for h in hosts if h.uuid != host.uuid]` so there's no risk of modifying the list while it's being iterated.

Reply:
i'll try